### PR TITLE
Support service type

### PR DESCRIPTION
**Pull request by rgstephens**
Based on issue #143 add support for service type for `rasax` and `rasa-production`

```yml
rasax:
  service:
    type: ClusterIP
    port: 5002
    nodePort: 
    loadBalancerIP:

versions:
  rasaProduction:
    type: ClusterIP
    port: 5005
    nodePort:
    loadBalancerIP:
```

### CODE REVIEW

File: charts/rasa-x/templates/rasa-services.yaml [hunk 1]
```diff
@@ -5,13 +5,22 @@ kind: "Service"
...
   labels:
-    {{ include "rasa-x.labels" $ | nindent 4 }}
+    {{- include "rasa-x.labels" $ | nindent 4 }}
     app.kubernetes.io/component: {{ .serviceName }}
 {{- with .service.annotations }}
   annotations:
     {{- toYaml . | nindent 4 }}
 {{- end }}
 spec:
+{{- if eq .serviceName "rasa-production" }}
```
Comment (tczekajlo) [inline charts/rasa-x/templates/rasa-services.yaml:15]: Why we do this only for rasa-production?

**Review by tczekajlo — APPROVED** [review]
@rgstephens  I added small things, the rest of it looks good to me 🎖️

File: charts/rasa-x/values.yaml [hunk 2]
```diff
@@ -12,8 +12,12 @@ rasax:
...
   tag: ""
-  # port on which Rasa X runs
-  port: 5002
+  # service
+  service:
+    # port on which Rasa X runs
+    port: 5002
+    # service type
+    type: ClusterIP
   # scheme by which Rasa X is accessible
```
Comment (tczekajlo) [inline charts/rasa-x/values.yaml:21]: You miss `rasax.port` that is used in the rasa-x-service template

**Review by tczekajlo — CHANGES_REQUESTED** [review]

File: charts/rasa-x/values.yaml (hunk 2 above)
Comment (rgstephens) [reply charts/rasa-x/values.yaml]: Thanks, dropped this

**Comment by tczekajlo** [comment]
Support for setting service type for Rasa X was added in the 4.2 version. The rasa production deployment is deprecated and not supported anymore, please use the dedicated chart for it (https://helm.rasa.com).

**Pull request closed without merging**